rdrf/registry/patients/parent_view.py

```diff
@@ -78,49 +57,58 @@ def dispatch(self, request, *args, **kwargs):
 
     def get(self, request, registry_code):
-        context = {}
         self.rdrf_context_manager = RDRFContextManager(self.registry)
-
-        patients_objects = self.parent.patient.all()
-        patients = []
-
-        forms_objects = (
-            RegistryForm.objects
-            .filter(registry=self.registry)
-            .exclude(is_questionnaire=True)
-            .order_by('position')
-        )
-
         progress = form_progress.FormProgress(self.registry)
+        fth = FormTitleHelper(self.registry, "")
 
-        for patient in patients_objects:
-            forms = []
-            for form in forms_objects:
-                if form.is_questionnaire or not request.user.can_view(form):
-                    continue
-                forms.append({
-                    "form": form,
-                    "progress": progress.get_form_progress(form, patient),
-                    "current": progress.get_form_currency(form, patient),
-                    "readonly": request.user.has_perm("rdrf.form_%s_is_readonly" % form.id)
-                })
-
-            rdrf_context = self.rdrf_context_manager.get_or_create_default_context(patient)
-            patients.append({
+        context = {
+            "parent": self.parent,
+            "patients": [{
                 "patient": patient,
                 "consent": consent_status_for_patient(registry_code, patient),
-                "context_id": rdrf_context.pk,
-                "forms": forms
-            })
+                "form_groups": [{
+                    "name": group_title,
+                    "forms": [{
+                        "form": form,
+                        "progress": progress.get_form_progress(form, patient),
+                        "current": progress.get_form_currency(form, patient),
+                        "readonly": request.user.has_perm("rdrf.form_%s_is_readonly" % form.id),
+                        "url": self._get_form_url(patient, form, cfg)
+                    } for form in forms]
+                } for cfg, group_title, forms in self._get_form_groups(patient)]
+            } for patient in self._get_parent_patients(self.parent)],
+            "registry_code": registry_code,
+            "form_titles": fth.all_titles_for_user(request.user)
+        }
 
-        context['parent'] = self.parent
-        context['patients'] = patients
-        context['registry_code'] = registry_code
+        return render(request, 'rdrf_cdes/parent.html', context)
 
-        fth = FormTitleHelper(self.registry, "")
-        context['form_titles'] = fth.all_titles_for_user(request.user)
+    def _get_parent_patients(self, parent):
+        for patient in parent.patient.all():
+            self.rdrf_context_manager.get_or_create_default_context(patient)
+            yield patient
+
+    def _get_form_url(self, patient, form, context_form_group=None):
+        if context_form_group:
+            if context_form_group.context_type == "F":
```

Comment:
A bit unrelated but can we add `is_fixed` and `is_multiple` properties to the ContextFormGroup model to avoid these kind of checks ?